rkt/run_prepared.go

```diff
@@ -125,6 +125,30 @@ func runRunPrepared(cmd *cobra.Command, args []string) (exit int) {
 		rktgid = -1
 	}
 
+	ovlOk := true
+	if err := common.SupportsOverlayPath(getDataDir()); err != nil {
+		if oerr, ok := err.(common.ErrOverlayUnsupported); ok {
+			stderr.Printf("disabling overlay support: %q", oerr.Error())
+			ovlOk = false
+		} else {
+			stderr.PrintE("error determining overlay support", err)
+			return 1
+		}
+	}
+
+	ovlPrep, err := p.overlayPrepared()
+	if err != nil {
+		stderr.PrintE("unable to determine prepared overlay state", err)
+		return 1
+	}
+
+	// should not happen, maybe the data directory moved from an overlay-enabled fs to another location
+	// between prepare and run-prepared
+	if ovlPrep && !ovlOk {
+		stderr.Print("unable to run prepared overlay-enabled pod: overlay not supported")
```

Comment:
I found this while history digging around here: https://github.com/coreos/rkt/pull/685

/cc @iaguis